xla/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -4578,6 +4582,322 @@ void FixDimsForRaggedOffset(std::vector<int64_t>& dims, int max_reg_per_batch) {
   dims[0] *= max_reg_per_batch;
 }
 
+class ScoreModFunc {
+  using Tensor_t = std::shared_ptr<cudnn_frontend::graph::Tensor_attributes>;
+  using Graph_t = std::shared_ptr<cudnn_frontend::graph::Graph>;
+
+ public:
+  ScoreModFunc(const xla::HloComputation* fwd_comp,
+               const xla::HloComputation* bwd_comp)
+      : fwd_comp_(fwd_comp), bwd_comp_(bwd_comp) {}
+
+  template <typename Functor>
+  absl::Status Initialize(cudnn_frontend::graph::Graph& graph,
+                          Functor* next_uid) {
+    TF_RETURN_IF_ERROR(UpdateHloParameterToCudnnMap(graph, fwd_hlo_to_cudnn_,
+                                                    fwd_comp_, next_uid));
+    TF_RETURN_IF_ERROR(
+        UpdateHloConstantToCudnnMap(graph, fwd_hlo_to_cudnn_, fwd_comp_));
+    if (bwd_comp_ != nullptr) {
+      TF_RETURN_IF_ERROR(
+          UpdateHloConstantToCudnnMap(graph, bwd_hlo_to_cudnn_, bwd_comp_));
+    }
+    return absl::OkStatus();
+  }
```

Comment:
Instead of having a separate constructor and an Initialize function, can you create a static factory function which either returns a `absl::StatusOr<ScoreModFunc>`?